frontend/src/concepts/pipelines/content/createRun/contentSections/RunTypeSectionScheduled.tsx

```diff
@@ -18,102 +26,129 @@ type RunTypeSectionScheduledProps = {
-        <b>Trigger type</b>
-      </Text>
-    </StackItem>
-    <StackItem>
-      <Radio
-        name="run-type-option-trigger-type"
-        label="Periodic"
-        isChecked={data.triggerType === ScheduledType.PERIODIC}
-        id={ScheduledType.PERIODIC}
-        onChange={() =>
-          onChange({ ...data, triggerType: ScheduledType.PERIODIC, value: DEFAULT_PERIODIC_OPTION })
-        }
-        body={
-          data.triggerType === ScheduledType.PERIODIC && (
-            <>
-              <Text>
-                <b>Run every</b>
-              </Text>
-              <SimpleDropdownSelect
-                options={Object.values(PeriodicOptions).map((v) => ({
-                  key: v,
-                  label: v,
-                }))}
-                value={data.value}
-                onChange={(value) => onChange({ ...data, value })}
-              />
-            </>
-          )
-        }
-      />
-    </StackItem>
-    <StackItem>
-      <Radio
-        name="run-type-option-trigger-type"
-        label="Cron"
-        id={ScheduledType.CRON}
-        isChecked={data.triggerType === ScheduledType.CRON}
-        onChange={() =>
-          onChange({
-            ...data,
-            triggerType: ScheduledType.CRON,
-            value: DEFAULT_CRON_STRING,
-          })
-        }
-        body={
-          data.triggerType === ScheduledType.CRON && (
-            <ClipboardCopy
-              hoverTip="Copy"
-              clickTip="Copied"
-              onChange={(value) => {
-                if (typeof value === 'string') {
-                  onChange({ ...data, value });
-                }
-              }}
-            >
-              {data.value}
-            </ClipboardCopy>
-          )
-        }
-      />
-    </StackItem>
-    <StackItem>
-      <Text>
-        <b>Duration</b>
-      </Text>
-    </StackItem>
-    <StackItem>
-      <RunTypeSectionDateTime
-        id="start-date"
-        label="Start date"
-        value={data.start}
-        onChange={(start) => onChange({ ...data, start })}
-      />
-    </StackItem>
-    <StackItem>
-      <RunTypeSectionDateTime
-        id="end-date"
-        label="End date"
-        value={data.end}
-        onChange={(end) => onChange({ ...data, end })}
-        adjustNow={(now) => {
-          if (data.start) {
-            const start = new Date(`${data.start.date} ${data.start.time}`);
-            start.setDate(start.getDate() + 7);
-            return start;
+const RunTypeSectionScheduled: React.FC<RunTypeSectionScheduledProps> = ({ data, onChange }) => {
+  const [periodicTime, setPeriodicTime] = React.useState<string>('');
+  const [periodicTimeUnit, setPeriodicTimeUnit] = React.useState<string>(PeriodicOptions.HOUR);
+
+  React.useEffect(() => {
+    onChange({
+      ...data,
+      value: periodicTime + periodicTimeUnit,
+    });
+    // only run when `periodicTime` & `periodicTimeUnit` changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I currently have to disable eslint here, but I'm open to suggestions for a more effective alternative if anyone can provide one.

Reply:
Why not parse the value from the data instead of storing local state?

Reply:
Absolutely avoid "copying" information in React. It's nothing but trouble. Single source of truth all data please.